src/core/openthread.cpp

```diff
@@ -66,9 +66,6 @@ static otDEFINE_ALIGNED_VAR(sInstanceRaw, sizeof(otInstance), uint64_t);
 otInstance *sInstance = NULL;
 #endif
 
-void OT_CDECL operator delete(void *, size_t) throw() { }
-void OT_CDECL operator delete(void *) throw() { }
```

Comment:
Thanks @jwhui, I was guessing that there must be something going on with Windows.

One idea that comes to my mind is to build Windows without new.hpp and letting Windows reuse standard library. For example, while building for Windows we could add DEFINES += NEW_HPP_ . Unfortunately, I have no way of trying that nor playing around Windows builds but the above syntax should work for Linux. This approach allows platform designer to control which new to use and when to link with standard libraries.

No matter what the fix is, we should not be addressing Windows specific issue by adding code into openthread.cpp.